spec/presenters/courses_presenter_spec.rb

```diff
@@ -13,6 +13,12 @@
       expect(subject.trained_count).to eq(1)
       expect(subject.trained_percent).to eq(50.0)
     end
+    context 'search campaign' do
+      it 'returns courses when searching' do
+        q = 'basket-weaving'
```

Comment:
This implicitly relies on the course title set by the factory ­— the "mystery guest" code smell, I think — so it's not clear just from looking at the test whether we expect it to pass or not.

Here we care about how the test interacts with the title, term and school, so you should set those explicitly for the course records used in this test.

It's also written against `nil` as the failure condition, but a search that returns no results will be an empty ActiveRecord relation, not `nil`, so this test will pass no matter what query you set. You can instead use `be_empty`. (It's very useful to make sure your tests fail in the expected ways when you change them, in addition to passing as written.)